### DISCUSSION TITLE
Translate Resource label and pluralized label - best pratice?

### DISCUSSION THREAD

**Discussion by remluben** [filamentphp/filament#1262] [Help] [answered]
Hi everyone,

when using translation in Filament I recently managed to use values from translation files for i.e. form fields for *MyResource*

```php
Forms\Components\TextInput::make('name')
  ->required()
  ->label(__('fields.my-resource.name')),
```

When it comes to translating the *label* and the *pluralized label* I did not find a way to use translation files out of the box. If specified in the global properties of a resource class it does not work, i.e

```php
// MyResource.php
protected static ?string $label = 'resources.my-resource';
protected static ?string $pluralLabel = 'resources.my-resources';
```

This outputs the values *resources.my-resource* and *resources.my-resources* instead of a translated version in the Filament admin panel, that's why I use to create a trait to use in my resources:

```php
<?php

namespace App\Filament\Resources\Concerns;

trait TranslatesLabels
{
    public static function getLabel(): string
    {
        return __(parent::getLabel());
    }

    public static function getPluralLabel(): string
    {
        return __(parent::getPluralLabel());
    }
}
```

I put this issue in the Q&A section, instead of making a pull request directly, because I am not sure if there's an out-of-the-box way to do it.

Is there an easier way to accomplish translation for resource label (and pluralized label)? Am I missing something?

Kind regards,
Benjamin

**Comment by danharrin** [comment, marked as answer]
Hi, this is how I recommend doing this - using the methods **instead** of properties.

**Reply by remluben** [reply]
Thanks a lot for your feedback.
